builtin/providers/aws/resource_aws_sns_topic.go

```diff
@@ -61,7 +62,7 @@ func resourceAwsSnsTopic() *schema.Resource {
 						log.Printf("[WARN] Error compacting JSON for Policy in SNS Topic")
 						return ""
 					}
-					value := normalizeJson(buffer.String())
+					value, _ := normalizeJsonString(buffer.String())
```

Comment:
This could be replaced with way simpler `StateFunc`, all these other code here is not really needed.